webui/react/src/pages/ExperimentDetails/ExperimentVisualization/LearningCurve.tsx

```diff
@@ -74,12 +77,58 @@ const LearningCurve: React.FC<Props> = ({
   const hasTrials = trialHps.length !== 0;
   const isExperimentTerminal = terminalRunStates.has(experiment.state as RunState);
 
-  const hyperparameters = useMemo(() => {
-    return fullHParams.reduce((acc, key) => {
-      acc[key] = experiment.hyperparameters[key];
+  const get_custom_search_varying_hps = (trialHps: TrialHParams[]) => {
```

Comment:
**suggestion:** this function looks like it is not dependent on anything inside the component `LearningCurve` and is purely reliant on the input parameter of `trialHps`. If that is the case, this function would be better to be defined just outside of the React component. It can be moved just above the start of `LearningCurve`.